curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-advanced-array-methods-by-building-a-statistics-calculator/65ca2d0625aa3a3201067f70.md

```diff
@@ -0,0 +1,168 @@
+---
+id: 65ca2d0625aa3a3201067f70
+title: Step 30
+challengeType: 0
+dashedName: step-30
+---
+
+# --description--
+
+Inside the `array.forEach()` callback function, check if the current element is inside the `counts` object.  If the element can be found, update the value of `counts[el]` by `1`. Otherwise, assign the number `0` to `counts[el]`.
+
+Below your `forEach`, log out the final result of `counts`.
+
+Then change your return statement to return `counts` instead of `array`.
```

Comment:
```suggestion
Below your `forEach`, log out the final result of `counts`. After that, change your return statement to return `counts` instead of `array`.
```